### ISSUE TITLE
Dry run update shows update to older package version

### ISSUE THREAD

**Issue by ijanos** [pypa/pipenv#1308]
Running update --dry-run shows update to a lower version of the pillow package. 

1. OS Type: Linux (Fedora 26)
1. Python version: 3.6.3
1. Pipenv version: 9.0.1

##### Actual result & Steps to replicate
```
$ mkdir test 
$ cd test/
$ pipenv install pillow
[...successful installation...]
$ pipenv update --dry-run --verbose
Checking dependencies…
pillow==4.3.0 is available (5.0.0 installed)!
```
Interestingly `pipenv update` re-installs 5.0.0

**Comment by Gifflen** [comment]
We've seem very similar behavior on linux and max on 3.6.1 and the latest pipenv.

**Comment by jacobian** [comment]
FWIW I can reproduce this with Django:

```
$ pipenv install django
...
Successfully installed django-2.0.2 pytz-2018.3
...

$ pipenv update --dry-run
Checking dependencies…
django==1.11.10 is available (2.0.2 installed)!
```

And just like the OP, it appears to just be `--dry-run` that does this:

```
$ pipenv update
...
All dependencies are now up-to-date!
$ pipenv graph
Django==2.0.2
  - pytz [required: Any, installed: 2018.3]
```

**Comment by kennethreitz** [comment]
update has been removed
